content/learn/book/ecs/_index.md

```diff
@@ -0,0 +1,140 @@
+
+`let foo.0 = bar` can seem opaque when you first read it, but in Rust it just means that you're assigning `bar` to the first field of the tuple struct `foo`.
+Using ordinary structs with named fields rather than tuple structs (even for simple components and resources!) can go a long way to improving the clarity of your code by providing more descriptive field names.
+
+### Smart pointers
+
+There are a few points to be aware of when working with "smart pointers" like components and resources:
+
+1. The value returned in a system when you ask for `Res<Score>` is not a plain `Score`, it is a `Res<Score>`. This is a resource that stores the score, similar to an ordinary `&` pointer, but with some special behavior (like change detection).
+2. Because `Res` and other smart pointers implement the `Deref` trait, Rust will attempt to automatically dereference it to match required types when it can.
+   1. This is commonly done when accessing fields and methods: calling `.title` on `Res<WindowDescriptor>` will grab the title field of `WindowDescriptor`, not the wrapping `Res<WindowDescriptor>` as the `Res` type has no field named `title`.
+3. When your types cannot be automatically dereferenced, you will need to manually dereference out of the outer smart pointer using `*`.
+   1. This is commonly needed when assigning a value to your data (rather than just one of its fields), or when using `match`.
+
+### Initializing structs with some default values
+
+Bevy often requires defining complex structs when creating entities with bundles or configuring the app with resources.
+In many cases, we want to use *some* default fields, but not all of them.
+As Rust does not have default arguments, we instead use [struct update syntax](https://doc.rust-lang.org/book/ch05-01-defining-structs.html#creating-instances-from-other-instances-with-struct-update-syntax) in combination with the [`default()`](https://doc.rust-lang.org/std/default/trait.Default.html) method from the `Default` trait to allow us to quickly set some values while leaving others unchanged.
+
+```rust
+// We want to quickly generate new enemies from this component bundle
+// By using some default fields
+#[derive(Bundle, Default)]
+struct EnemyBundle {
+	enemy: Enemy,
+	life: Life,
+	attack: Attack,
+	defense: Defense,
+}
+
+// Dataless unit structs automatically have a default value
+// Since only one possible value exists
+struct Enemy;
+
+struct Life(u32);
+
+// Manually implementing Default for one of our component types
+// allowing us to control the starting value
+impl Default for Life {
+	fn default() -> Self {
+		Life(100)
+	}
+}
+
+// These components will have a default value of 0, as u32::default() == 0
+#[derive(Default)]
+struct Defense(u32);
+
+#[derive(Default)]
+struct Attack(u32);
+
+let my_enemy = EnemyBundle {
+	attack: 10,
+	defense: 5,
+	// All other fields are initialized from their default value
+	// setting enemy: Enemy and life: Life(100)
+	..Default::default()
+}
+```
```

Comment:
The example should just focus on the Struct Update Syntax, not bleed into Bevy features. Here the concept of _bundles_ is added, unnecessarily increasing complexity (especially since bundles are not introduced yet). A pure Rust example would be much more well understood.

Basically, just show the definition of a struct with many fields (not too many or the code explodes in length; 4 fields are enough). Say that the struct implements `Default` (don't show the full impl though, it's too much code and it's irrelevant), and then create an instance of the struct by filling a couple of fields and then use `..Default::default()`

Reply:
Agreed: this has been simplified and I left a note below to talk about its relevance to bundles.